### PR TITLE
Experimental support for Google-style color bitmap fonts

### PR DESCRIPTION
**Pull request by rsheeter**
Copying sample code manually into #395 is kind of annoying, creating PR instead.

### CODE REVIEW

File: Lib/fontTools/subset.py [hunk 1]
```diff
@@ -1254,6 +1254,23 @@ def collect_features(self):
     return _uniq_sort(sum((s.Script.collect_features()
                            for s in self.ScriptRecord), []))
 
+# CBLC will inherit it
+@_add_method(ttLib.getTableClass('EBLC'))
+def subset_glyphs(self, s):
+  for strike in self.strikes:
+    for indexSubTable in strike.indexSubTables:
+      indexSubTable.names = [n for n in indexSubTable.names if n not in s.glyphs]
```
Comment (anthrotype) [inline Lib/fontTools/subset.py]: @rsheeter isn't there a contradiction here? `s.glyphs` is meant to be the list of glyph names which one wants to *retain*. So here you would want to keep all the names which are in `s.glyphs`, and not those which are not...

```diff
-indexSubTable.names = [n for n in indexSubTable.names if n not in s.glyphs]
+indexSubTable.names = [n for n in indexSubTable.names if n in s.glyphs]
```

File: Lib/fontTools/subset.py [hunk 2]
```diff
@@ -1254,6 +1254,23 @@ def collect_features(self):
...
+    strike.indexSubTables = [i for i in strike.indexSubTables if i.names]
+  self.strikes = [s for s in self.strikes if s.indexSubTables]
+
+  return True
+
+# CBDC will inherit it
+@_add_method(ttLib.getTableClass('EBDT'))
+def subset_glyphs(self, s):
+  self.strikeData = [_dict_subset(strike, s.glyphs) for strike in self.strikeData]
```
Comment (anthrotype) [inline Lib/fontTools/subset.py]: this raises `KeyError` whenever a glyph name is in the list of glyphs to retain `s.glyphs` but is not also among the keys of the `strike` dictionary.

For example, in the example color fonts from:
https://github.com/googlei18n/color-emoji/tree/master/specification/examples

the glyph '.notdef' is not contained in the CBDT strikeData, but is contained in the post table, so the `_dict_subset` function fails with KeyError.

I think you should subset the CBDT table's strikeData like that:

```python
self.strikeData = [{g: strike[g] for g in s.glyphs if g in strike}
                   for strike in self.strikeData]
```

**Comment by anthrotype** [comment]
@rsheeter don't know why github is saying the your pull request branch has merge conflicts (i tried merging locally and it worked..)
could you maybe try rebasing on behdad/master?

**Comment by anthrotype** [comment]
I guess it's because the subset.py module has been renamed to subset/__init__.py..
Also, before we merge this PR, it'd be nice if we add some unit tests, based on the ones @miguelsousa wrote in https://github.com/behdad/fonttools/pull/411.

**Comment by rsheeter** [comment]
Rebased & added a test.

**Comment by anthrotype** [comment]
excellent, thanks! :+1:

**Pull request merged**